ArduPlane/control_modes.cpp

```diff
@@ -150,12 +157,30 @@ uint8_t Plane::readSwitch(void) const
 {
     uint16_t pulsewidth = RC_Channels::get_radio_in(g.flight_mode_channel - 1);
     if (pulsewidth <= 900 || pulsewidth >= 2200) return 255;            // This is an error condition
-    if (pulsewidth <= 1230) return 0;
-    if (pulsewidth <= 1360) return 1;
-    if (pulsewidth <= 1490) return 2;
-    if (pulsewidth <= 1620) return 3;
-    if (pulsewidth <= 1749) return 4;              // Software Manual
-    return 5;                                                           // Hardware Manual
+
+    if (g2.extra_modes == 0) {
+        // Old ranges, for backwards compatibility.
+        if (pulsewidth <= 1230) return 0;
```

Comment:
Since you're touching these, can you please make them all match [RC_Channel/RC_Channel.cpp](https://github.com/ArduPilot/ardupilot/blob/master/libraries/RC_Channel/RC_Channel.cpp#L413-L433) by using ```pulsewidth < 1231``` instead of ```pulsewidth <= 1230```.

Reply:
Done!